src/script/components/Modals/PrimaryModal/PrimaryModal.tsx

```diff
@@ -307,10 +329,15 @@ export const PrimaryModalComponent: FC = () => {
                     type="password"
                     autoComplete="off"
                     value={passwordValue}
+                    ref={passwordValueRef}
                     onChange={event => setPasswordValue(event.currentTarget.value)}
+                    pattern={ValidationUtil.getNewPasswordPattern(Config.getConfig().NEW_PASSWORD_MINIMUM_LENGTH)}
+                    markInvalid={isFormSubmitted && !passwordValueRef.current?.validity.valid}
```

Comment:
Question, it seems like other inputs have their border-color set to red when invalid, could you check why it's not the case here? We might need to add a last bit of css

Reply:
Nice catch 🦸 
There was some wrapper css that is overriding the default behavior. I realized we don't event need that css.
For the sake of consistency, we will add this css on UI toolkit if we still need it.